mongodbatlas/resource_mongodbatlas_cluster.go

```diff
@@ -1175,7 +1178,19 @@ func expandReplicationSpecs(d *schema.ResourceData) ([]matlas.ReplicationSpec, e
 					id = oldID
 				}
 			}
-			regionsConfig, err := expandRegionsConfig(spec["regions_config"].(*schema.Set).List())
+
+			replaceRegion := ""
+			originalRegion := ""
+
+			if okPRName && d.Get("provider_name").(string) == "GCP" && cast.ToString(d.Get("cluster_type")) == "REPLICASET" {
```

Comment:
[q] What would happen when the type is `SHARDED`?

Reply:
if the provider is not GCP and the cluster type is not replicate set(since its automated somehow in api response) with the user config, it will not update the region name in replication specs as it commented in ticket

Reply:
Thanks!